lib/net/peer.js

```diff
@@ -2235,6 +2344,28 @@ class PeerOptions {
   static getRate(hash) {
     return -1;
   }
+
+  /**
+   * Get the peer type by address.
+   * See PoolOptions.peerTypes.
+   * @private
+   * @param {NetAddress}
+   * @returns {Number}
+   */
+
+  static getTypeByAddr(addr) {
```

Comment:
Why do this and `static getPeerType()` exist?

Reply:
It follows the same pattern of the other `static` functions, to set the initial values of the class methods. These class methods are then overwritten by the functions that are passed in.